### PR TITLE
Add clear_*() functions

### PR DESCRIPTION
**Pull request by sgallagher**
For any place where we have a .add_*() function, add a corresponding
.clear_*() function.

Fixes: https://github.com/fedora-modularity/libmodulemd/issues/294

Signed-off-by: Stephen Gallagher <sgallagh@redhat.com>

### CODE REVIEW

File: modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h [hunk 1]
```diff
@@ -100,6 +100,19 @@ modulemd_dependencies_set_empty_buildtime_dependencies_for_module (
   ModulemdDependencies *self, const gchar *module_name);
 
 
+/**
+ * modulemd_dependencies_clear_buildtime_dependencies:
+ * @self: This #ModuleDependencies
```
Comment (mmathesius) [inline modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h]: Should be Module**md**Dependencies. There are four other instances of this same typo--all in this file.

File: modulemd/v2/include/modulemd-2.0/modulemd-buildopts.h [hunk 2]
```diff
@@ -118,6 +118,18 @@ modulemd_buildopts_remove_rpm_from_whitelist (ModulemdBuildopts *self,
...
+/**
+ * modulemd_buildopts_clear_whitelist:
+ * @self: This #ModulemdBuildopts
+ *
+ * Remove all RPMs from the whitelist.
+ *
+ * Since: 2.5
+ */
+void
+modulemd_buildopts_clear_whitelist (ModulemdBuildopts *self);
```
Comment (mmathesius) [inline modulemd/v2/include/modulemd-2.0/modulemd-buildopts.h]: For naming consistency, would it be more appropriate to name the new function `modulemd_buildopts_clear_rpm_whitelist()`?

File: modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h (hunk 1 above)
Comment (mmathesius) [reply modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h]: Actually, running `grep -r '#Module[A-Z]' ...` on the source directory reveals several more similar typos.

File: modulemd/v2/include/modulemd-2.0/modulemd-buildopts.h (hunk 2 above)
Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-buildopts.h]: Yeah, probably best. Thanks

File: modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h (hunk 1 above)
Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h]: I'll open an issue and fix that more globally, then. It's a documentation comment, so it's not particularly urgent.

Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-dependencies.h]: I opened #301 to track that.

File: modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h [hunk 3]
```diff
@@ -517,6 +567,18 @@ modulemd_module_stream_v1_remove_rpm_api (ModulemdModuleStreamV1 *self,
...
+/**
+ * modulemd_module_stream_v1_clear_rpm_api:
+ * @self: (in): This #ModulemdModuleStreamV1 object.
+ *
+ * Remove all RPMs from the list of stable public API.
+ *
+ * Since: 2.5
+ */
+void
+modulemd_module_stream_v1_clear_rpm_api (ModulemdModuleStreamV1 *self);
```
Comment (mmathesius) [inline modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h]: I'm torn on this, but for naming consistency it feels like function name should be in the plural form, `*_clear_rpm_apis()`.

File: modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h [hunk 4]
```diff
@@ -756,6 +842,32 @@ modulemd_module_stream_v1_remove_runtime_requirement (
...
+/**
+ * modulemd_module_stream_v1_clear_buildtime_requirement
+ * @self: (in): This #ModulemdModuleStreamV1 object.
+ *
+ * Remove all buildtime dependencies for this module.
+ *
+ * Since: 2.5
+ */
+void
+modulemd_module_stream_v1_clear_buildtime_requirement (
```
Comment (mmathesius) [inline modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h]: Again, this and the next function feel like they should be named in the plural form.

File: modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h (hunk 3 above)
Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h]: I struggled with that, but in the other cases, the plural was more obvious on the last word. In this case, I think the singular fits better.
As long as the verb (create) is consistent, I think it's probably fine.

File: modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h (hunk 4 above)
Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h]: This one was an oversight. I'll fix that right up.

Comment (sgallagher) [reply modulemd/v2/include/modulemd-2.0/modulemd-module-stream-v1.h]: Fixed and force-pushed.

**Comment by mmathesius** [comment]
I'm guessing `modulemd_component_rpm_reset_arches()` and `modulemd_component_rpm_reset_multilib_arches()` are _not_ going to get renamed to `modulemd_component_rpm_clear_arches()` and `modulemd_component_rpm_clear_multilib_arches()`?

**Comment by mmathesius** [comment]
And `modulemd_defaults_v1_set_empty_default_profiles_for_stream()` is likely not going to get renamed to `modulemd_defaults_v1_clear_default_profiles_for_stream()`?

I'm also guessing there's little practical use for implementing the following:
* `modulemd_module_clear_translations()`
* `modulemd_module_index_clear_module_streams()`                                       
* `modulemd_module_index_clear_defaults()`                                               
* `modulemd_module_index_clear_translations()`

**Comment by mmathesius** [comment]
Aside from my above queries about documentation related typos, function naming, and completeness, the actual code LGTM.

**Comment by sgallagher** [comment]
> I'm guessing `modulemd_component_rpm_reset_arches()` and `modulemd_component_rpm_reset_multilib_arches()` are _not_ going to get renamed to `modulemd_component_rpm_clear_arches()` and `modulemd_component_rpm_clear_multilib_arches()`?

Correct, these were pre-existing and I don't want to break compatibility. If you feel strongly that they should have matching names, I can add an alias for them, but I'm not sure it's worth it.

I'll defer to your judgement as reviewer here.

> And `modulemd_defaults_v1_set_empty_default_profiles_for_stream()` is likely not going to get renamed to `modulemd_defaults_v1_clear_default_profiles_for_stream()`?
> 

Same as above.

> I'm also guessing there's little practical use for implementing the following:
> 
> * `modulemd_module_clear_translations()`
> * `modulemd_module_index_clear_module_streams()`
> * `modulemd_module_index_clear_defaults()`
> * `modulemd_module_index_clear_translations()`

These aren't being added because it's actually prohibitively difficult. They are inter-related and as such it's better and safer to rely on [`modulemd_module_index_remove_module()`](https://fedora-modularity.github.io/libmodulemd/latest/modulemd-2.0-Modulemd.ModuleIndex.html#modulemd-module-index-remove-module) instead.

**Comment by mmathesius** [comment]
> Correct, these were pre-existing and I don't want to break compatibility. If you feel strongly that they should have matching names, I can add an alias for them, but I'm not sure it's worth it.
> 
> I'll defer to your judgement as reviewer here.

I was mostly looking for confirmation here. As long as the functions that provide the functionality already exist, adding aliases would just be an extra burden. (Unless, of course, there is an outcry for this from the consumers of the API!)

> > And `modulemd_defaults_v1_set_empty_default_profiles_for_stream()` is likely not going to get renamed to `modulemd_defaults_v1_clear_default_profiles_for_stream()`?
> 
> Same as above.
> 
> > I'm also guessing there's little practical use for implementing the following:
> > 
> > * `modulemd_module_clear_translations()`
> > * `modulemd_module_index_clear_module_streams()`
> > * `modulemd_module_index_clear_defaults()`
> > * `modulemd_module_index_clear_translations()`
> 
> These aren't being added because it's actually prohibitively difficult. They are inter-related and as such it's better and safer to rely on [`modulemd_module_index_remove_module()`](https://fedora-modularity.github.io/libmodulemd/latest/modulemd-2.0-Modulemd.ModuleIndex.html#modulemd-module-index-remove-module) instead.

Agreed.

**Pull request merged**